src/Query/Filter/HappensDateField.ts

```diff
@@ -70,6 +73,31 @@ export class HappensDateField extends Field {
                     };
                     relative = ' on';
                 }
+
+                if (happensMatch[1] === 'in' && matchCurrentPeriod !== null) {
+                    filterFunction = (task: Task) => {
+                        return this.dates(task).some(
+                            (date) => date && DateField.isDateInCurrentPeriod(date, matchCurrentPeriod[1]),
+                        );
```

Comment:
This is weird, I'm in a Monday-Sunday locale, I will dig into this, thanks

Reply:
I think that using `moment().startOf('isoWeek')` instead of `week` should do it. 

https://momentjs.com/docs/#/manipulating/start-of/

I found it via various StackOverflow answers, such as
https://stackoverflow.com/questions/34862870/moment-js-issue-isoweekyear-isoweek-method-behavior